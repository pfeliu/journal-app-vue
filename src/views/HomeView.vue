<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <button class="btn btn-primary" @click="gotoDaybook">Go to Daybook</button>
  </div>
</template>

<script>
export default {
  methods: {
    gotoDaybook() {
      this.$router.push({name: 'no-entry'})
    }
  }
}
</script>

