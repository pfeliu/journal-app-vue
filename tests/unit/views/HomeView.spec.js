import HomeView from '@/views/HomeView.vue';
import { shallowMount } from '@vue/test-utils';

describe('Home view', () => {
  test('should match with snapshot', () => {
    const wrapper = shallowMount(HomeView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should click and redirect', () => {
    const mockRouter = { push: jest.fn() };

    const wrapper = shallowMount(HomeView, {
      global: {
        mocks: { $router: mockRouter },
      },
    });

    wrapper.find('button').trigger('click');

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
  });
});
