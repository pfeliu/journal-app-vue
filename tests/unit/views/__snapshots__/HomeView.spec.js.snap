// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view should match with snapshot 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn btn-primary">Go to Daybook</button></div>`;
